### ISSUE TITLE
XML tag with `xml:space="preserve"` containing only spaces should not be collapsed

### ISSUE THREAD

**Issue by ivangreene** [Ekryd/sortpom#402]
sortpom is suggesting to collapse an XML tag that contains only space characters, even though it has the `xml:space="preserve"` attribute on it. This is used to configure some plugins, e.g: https://www.habarta.cz/typescript-generator/maven/typescript-generator-maven-plugin/generate-mojo.html#indentString

The diff when trying to use this:
```diff
[ERROR]     pom.xml
[ERROR]         @@ -110,7 +110,7 @@
[ERROR]          ··········<classesFromAutomaticJaxrsApplication>true</classesFromAutomaticJaxrsApplication>
[ERROR]          ··········<outputFile>${generated-ts-client-file}</outputFile>
[ERROR]          ··········<mapEnum>asEnum</mapEnum>
[ERROR]         -··········<indentString·xml:space="preserve">··</indentString>
[ERROR]         +··········<indentString·xml:space="preserve"/>
[ERROR]          ··········<noFileComment>true</noFileComment>
[ERROR]          ··········<extensions>
[ERROR]          ············<extension>cz.habarta.typescript.generator.ext.AxiosClientExtension</extension>
```

(Btw I am using this from spotless)

**Comment by Ekryd** [comment]
I believe that I need to check the XML framework that I use to see this specific behaviour 👀

**Comment by Ekryd** [comment]
Wow! You found a bug in the plugin. Great Job!

When I get the XML from the SAXReader, I get the XML tags and the empty whitespace between the tags. In order to identify  the empty lines I use a regex: `[\t ]*\r?\n?[\t ]*` that unfortunately also identified your two spaces. Your text was removed and you got the empty tag.

I have changed the regex to the following: `[\t ]*(\r|\n|\r\n)[\t ]*` which solves your problem. but if somebody has the need to set a newline as text in an XML tag with xml:space="preserve" they will encounter the same problem.

Currently, I'm seeing if I get the SAXParser to only return XML tags and any text that is directly wrapped by those tags.

**Comment by Ekryd** [comment]
@ivangreene, I released new version of the plugin `3.4.1`, that should solve this issue. Please try it out and tell me if it solves your problem.

I added a test that should test several whitespace cases combined with preserve:
https://github.com/Ekryd/sortpom/blob/master/sorter/src/test/resources/PreserveContent_input.xml
https://github.com/Ekryd/sortpom/blob/master/sorter/src/test/resources/PreserveContent_output.xml
https://github.com/Ekryd/sortpom/blob/master/sorter/src/test/java/sortpom/sort/LineSeparatorTest.java#L39
